x/mongo/driver/connstring/connstring.go

```diff
@@ -86,8 +86,8 @@ type ConnString struct {
 	WNumber                            int
 	WNumberSet                         bool
 	Username                           string
-	ZlibLevel                          int
-	ZlibLevelSet                       bool
+	ZlibLevel                          *int
```

Comment:
I think we should keep these as `int`s and keep the `*Set` fields so we can keep this whole thing consistent. I'd rather not chance the way we do things halfway through.

Reply:
This was done so that we can differentiate between the user not setting a level and setting a level of 0 (currently the default is not 6 as expected). Were you thinking of a different approach?

Reply:
The `ZlibLevelSet` field differentiates whether the user has set `ZlibLevel`. Basically, we're representing the same information using two fields instead of one. Even though they represent the same information I'd like to keep these fields consistent with the rest of the connection string fields.

Reply:
I reverted this change and made the zstd setting consistent